signer/tests/integration/transaction_signer.rs

```diff
@@ -250,6 +255,137 @@ async fn signer_rejects_stacks_txns_with_too_high_a_fee(
+        .with_first_bitcoin_core_client()
+        .with_mocked_emily_client()
+        .with_mocked_stacks_client()
+        .build();
+
+    // We need this or the contract call will fail validation with an
+    // unrelated error, since we mock reaching out to the stacks node.
+    set_deposit_incomplete(&mut ctx).await;
+
+    let (rpc, faucet) = sbtc::testing::regtest::initialize_blockchain();
+
+    // This confirms a deposit transaction, and has a nice helper function
+    // for storing a real deposit.
+    let mut setup = TestSweepSetup::new_setup(rpc, faucet, 10000, &mut rng);
+
+    // Let's get the blockchain data into the database.
+    let chain_tip = BitcoinBlockRef {
+        block_hash: setup.sweep_block_hash.into(),
+        block_height: setup.sweep_block_height,
+    };
+    backfill_bitcoin_blocks(&db, rpc, &chain_tip.block_hash).await;
+
+    // This is all normal things that need to happen in order to pass
+    // validation.
+    setup.store_happy_path_data(&db).await;
+
+    let (mut req, _) = crate::complete_deposit::make_complete_deposit(&setup);
+
+    req.deployer = ctx.config().signer.deployer;
+    let network = InMemoryNetwork::new();
+    let mut tx_signer = TxSignerEventLoop {
+        network: network.connect(),
+        context: ctx.clone(),
+        context_window: 10000,
+        wsts_state_machines: LruCache::new(NonZeroUsize::new(100).unwrap()),
+        signer_private_key: setup.aggregated_signer.keypair.secret_key().into(),
+        threshold: 2,
+        rng: rand::rngs::StdRng::seed_from_u64(51),
+        dkg_begin_pause: None,
+        dkg_verification_state_machines: LruCache::new(NonZeroUsize::new(5).unwrap()),
+        stacks_sign_request: LruCache::new(NonZeroUsize::new(2).unwrap()),
+    };
+
+    // We need this so that there is a live "network". Otherwise will error when
+    // trying to send a message at the end.
+    let _rec = ctx.get_signal_receiver();
+
+    // We need the wallet to get the real txid to pass the final checks
+    let wallet = SignerWallet::load(&ctx, &chain_tip.block_hash)
+        .await
+        .unwrap();
+
+    // We can sign a transaction generated by a coordinator who is not in
+    // the signer set, so the origin doesn't matter much for this function
+    // call.
+    let origin_public_key: PublicKey = Faker.fake_with_rng(&mut rng);
+
+    // Let's create a proper sign request.
+    let mut request = StacksTransactionSignRequest {
+        aggregate_key: setup.aggregated_signer.keypair.public_key().into(),
+        contract_tx: ContractCall::CompleteDepositV1(Box::new(req.clone())).into(),
+        nonce: 1,
+        tx_fee: 100_000,
+        txid: Faker.fake_with_rng::<StacksTxId, _>(&mut rng).into(),
+    };
+    wallet.set_nonce(request.nonce);
+    request.txid = MultisigTx::new_tx(&request.contract_tx, &wallet, request.tx_fee)
+        .tx()
+        .txid();
+
+    // In this test we use `handle_stacks_transaction_sign_request` instead of
```

Comment:
This comment confused me more than it helped 😅 I'd just write "simulate handling the stacks signing request which hasn't been processed yet, which will store its id in the 'already-signed' hashset for the chain tip" or something.

Reply:
Yeah, that wasn't really helpful, simplified a bit in [cf040e6](https://github.com/stacks-sbtc/sbtc/pull/1691/commits/cf040e643281201bb1fa9bf958b519091caaf55e)